### PR TITLE
gatsby-remark-prismjs: support inline code (single ticks)

### PR DESCRIPTION
**Pull request by docwhat**
Currently, the gatsby-remark-prismjs plugin is ignoring inline code markdown
(i.e. text inside single backticks).  This means that most stock prismjs CSS
and even the typography prismjs themes don't work.

The specific CSS selectors to style inline code element usually looks like
`:not(pre) > code[class*="language-"]`

By setting the `inlineCodeMarker` option, you can then do:

``` markdown
Search for the `css›.some-class { color: red }` code and change `red` to
`blue`.
```

Which is converted into this before being passed to prismjs:

``` html
Search for the <code class="language-css">.some-class { color: red }</code>
code and change <code>red</code> to <code>blue</code>.
```

### CODE REVIEW

File: packages/gatsby-remark-prismjs/README.md [hunk 1]
```diff
@@ -27,6 +27,13 @@ plugins: [
             // This is an uncommon use-case though;
             // If you're unsure, it's best to use the default value.
             classPrefix: "language-",
+            // This is used to allow setting a language for inline code
```
Comment (KyleAMathews) [inline packages/gatsby-remark-prismjs/README.md]: Could you add an example of how this works? I had to read the code before I realized that the `inlineCodeMarker` let you do something like `javascript>let p = "blue"`

Comment (docwhat) [reply packages/gatsby-remark-prismjs/README.md]: Sure. Gimme a second.

Comment (docwhat) [reply packages/gatsby-remark-prismjs/README.md]: Done!

**Comment by KyleAMathews** [comment]
Thanks!

Looks like Travis is failing now

**Comment by docwhat** [comment]
> Looks like Travis is failing now

Yeah... that's my bad. I did a regression because I'm bouncing between two laptops and that fix was on the other one.

**Comment by m-allanson** [comment]
Good stuff, thanks @docwhat!

**Comment by KyleAMathews** [comment]
This is breaking some pages on gatsbyjs.org e.g. tutorial part 2 so reverting this for now. This page breaks at some html written in an inline code element `<button>`.

screen shot 2018-03-17 at 5 51 13 pm

**Comment by docwhat** [comment]
What a pain. That’s the language-none class being applied. 

I’m confused though. Is then button inside the inlineCode element? Or something else?

Ideally if I could have a test for it I could fix it.

**Comment by KyleAMathews** [comment]
Yeah, the markdown just looks like ``text text `<button>` text``.

**Comment by docwhat** [comment]
The problem is that the content of the `innerCode` is not HTML escaped.  From the test's snippet:

```
   "type": "html",
   "value": "<code class=\\"language-none\\"><button /></code>",
```

**Comment by docwhat** [comment]
This isn't actually a bug in my code... it's a bug in the way we're using PrismJS:  The `language-none` (or `language-whatever` as long as it doesn't exist) turns the PrismJS highlighter into a noop; The inner text isn't escaped or encoded.

So this:

```none
Blocks have the <button /> bug as well.
```

Renders into:

```html
<div class="gatsby-highlight">
  <pre class="language-none"><code class="language-none">Blocks have the </code><button><code class="language-none"> bug as well.</code></button></pre>
 </div>
```

Like your picture, the `<button>` element wraps the second text element (i.e. " bug as well").

I feel like that the default language should be something like `plain` or `text`, a language that doesn't exist in Prism, that would do the HTML encoding.

**Comment by docwhat** [comment]
I have fixed both issues (and resubmitted this code) is PR #4599 

@KyleAMathews and @m-allanson I'd appreciate your feedback on my fix(es).

**Pull request merged**